src/tools/runk/libcontainer/src/builder.rs

```diff
@@ -48,6 +48,16 @@ impl Container {
             }
         }
 
+        if let Some(process) = spec.process.as_ref() {
+            // runk always launches containers with detached mode, so users have to
+            // use a console socket with run or create operation when a terminal is used.
+            if process.terminal && self.console_socket.is_none() {
+                return Err(anyhow!(
+                    "cannot allocate a pseudo-TTY without setting a console socket"
```

Comment:
Could you add a unit test for this scenario?

Reply:
@jodh-intel Thank you for the review. I added the unit test.